### PR TITLE
Refactoring the translation process

### PR DESCRIPTION
**Pull request by pdcp1**
### This PR 
- Fixes https://github.com/responsible-ai-collaborative/aiid/issues/2701
- Add a migration to delete all translation placeholders `translated-xx-`
- Move the translation logic to a new "Translate Reports" GitHub action
  - This workflow runs **every day at 8 AM**
- Add a translation status badge on the Readme page

badge

### Deployment
We need to make sure that these env variables and secrets are setup in both environments `staging` and `production`:
- `secrets.MONGODB_TRANSLATIONS_CONNECTION_STRING`
- `secrets.GOOGLE_TRANSLATE_API_KEY`
- `vars.TRANSLATE_DRY_RUN`

### Example executions
- Pablo's environment https://github.com/pdcp1/aiid/actions/workflows/translate.yml

### CODE REVIEW

File: .github/workflows/translate.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Translate Reports
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 8 * * *" # Run every day at 8 AM GMT
+
+jobs:
+  translate:
```
Comment (kepae) [inline .github/workflows/translate.yml:9]: I believe you will need to specify the environment for the job. e.g. from other workflow files, we use:
`environment: ${{ inputs.environment }}`

This is intended for production? 
In staging, can we configure the workflow to have the placeholders?

Comment (kepae) [reply .github/workflows/translate.yml:9]: (We can start by getting this migrated as-is first. :-))

Comment (pdcp1) [reply .github/workflows/translate.yml:12]: done!

File: site/gatsby-site/src/scripts/translateReports.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const Translator = require('./Translator');
+
+const { getLanguages } = require('../../i18n');
+
+const reporter = { log: console.log, error: console.error, warn: console.warn };
+
+(async () => {
+  let mongoClient;
+
+  try {
```
Comment (kepae) [inline site/gatsby-site/src/scripts/translateReports.js:18]: nitpick: move the log above the try. Let's keep `try` as close as possible to failing logic.

File: site/gatsby-site/src/scripts/translateReports.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      reporter,
+    });
+
+    // Run the translation process
+    await translator.run();
+
+    console.log('Translation completed successfully.');
+  } catch (error) {
+    console.error('Error during the translation process:', error.message);
+    process.exit(1);
```
Comment (kepae) [inline site/gatsby-site/src/scripts/translateReports.js:51]: there are other fatal errors above where you return early. Would you like to exit with failure there as well? That might be best to surface errors to github actions. Please advise.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -97,11 +97,9 @@ jobs:
           GATSBY_ALGOLIA_SEARCH_KEY: ${{ vars.GATSBY_ALGOLIA_SEARCH_KEY }}
           GATSBY_AVAILABLE_LANGUAGES: ${{ vars.GATSBY_AVAILABLE_LANGUAGES }}
           GATSBY_REALM_APP_ID: ${{ vars.GATSBY_REALM_APP_ID }}
-          GOOGLE_TRANSLATE_API_KEY: ${{ secrets.GOOGLE_TRANSLATE_API_KEY }}
           MONGODB_CONNECTION_STRING: ${{ secrets.MONGODB_CONNECTION_STRING }}
           MONGODB_REPLICA_SET: ${{ secrets.MONGODB_REPLICA_SET }}
           MONGODB_TRANSLATIONS_CONNECTION_STRING: ${{ secrets.MONGODB_TRANSLATIONS_CONNECTION_STRING }}
```
Comment (kepae) [inline .github/workflows/deploy.yml:102]: can also remove `MONGO_TRANSLATIONS_CONNECTION_STRING` too right? here and in other files.

**Review by kepae — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (pdcp1) [reply .github/workflows/deploy.yml:102]: In fact, that environment variable is utilized in other locations.
For example, the `MONGODB_TRANSLATIONS_CONNECTION_STRING` is loaded into `config.mongodb.translationsConnectionString` and it's used in
- Algolia index update process https://github.com/responsible-ai-collaborative/aiid/blob/4b5603012c380a6beac122717d4b6b48b2df5218/site/gatsby-site/src/scripts/algolia-update.js#L23
- In the TSNE update script https://github.com/responsible-ai-collaborative/aiid/blob/4b5603012c380a6beac122717d4b6b48b2df5218/site/gatsby-site/src/utils/updateTsne.js#L10
- In the `createDownloadIndexPage` https://github.com/responsible-ai-collaborative/aiid/blob/4b5603012c380a6beac122717d4b6b48b2df5218/site/gatsby-site/page-creators/createDownloadIndexPage.js#L14

File: .github/workflows/translate-production.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+name: Translate Reports - Production
+
+on:
+  schedule:
+    - cron: "0 5 * * *" # Run every day at 5 AM GMT
+  
+  workflow_dispatch:
+
+jobs:
+  call-deploy:
```
Comment (cesarvarela) [inline .github/workflows/translate-production.yml:10]: this should be called `call-translate` or similar

File: .github/workflows/translate-staging.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: Translate Reports - Staging
+
+on:
+  schedule:
+    - cron: "0 5 * * *" # Run every day at 5 AM GMT
+  
+  workflow_dispatch:
+
+jobs:
+  call-deploy:
```
Comment (cesarvarela) [inline .github/workflows/translate-staging.yml:10]: ditto

File: .github/workflows/translate-production.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+name: Translate Reports - Production
+
+on:
+  schedule:
+    - cron: "0 5 * * *" # Run every day at 5 AM GMT
+  
+  workflow_dispatch:
```
Comment (cesarvarela) [inline .github/workflows/translate-production.yml:7]: This is to enable manual triggering or it is a leftover?

File: .github/workflows/translate.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Install NPM dependencies
+        run: npm ci
+        working-directory: site/gatsby-site
+
+      - name: Run translation script
+        run: node src/scripts/translateReports.js
```
Comment (cesarvarela) [inline .github/workflows/translate.yml:30]: nit. I prefer using package.json scripts, which makes it easier to add/remove parameters, etc.

**Review by cesarvarela — APPROVED** [review]

File: .github/workflows/translate-production.yml (hunk 7 above)
Comment (pdcp1) [reply .github/workflows/translate-production.yml:7]: Yes, it is to enable manual triggering if it's needed

File: .github/workflows/translate.yml (hunk 8 above)
Comment (pdcp1) [reply .github/workflows/translate.yml]: done!

**Review by cesarvarela — COMMENTED** [review]
Is this test failing expected?

 Translation Badges › Should be visible on an report card on the citation page if it was translated

**Comment by pdcp1** [comment]
> Is this test failing expected?
> 
> Translation Badges › Should be visible on an report card on the citation page if it was translated

I just moved into `e2e-full` folder and now it uses the in-memory database. It passes ✅ 
@cesarvarela please review

**Review by cesarvarela — APPROVED** [review]
🥳

**Comment by kepae** [comment]
@pdcp1 Looks good to go for the next release? 🚀 ?

**Pull request merged**